File: python/etc/prime_num.py
import math

def is_prime_number(x):
  if x == 1:
    return False
  for i in range(2, int(math.sqrt(x)) + 1):
    if x % i == 0:
      return False
  return True

print(is_prime_number(7))
print(is_prime_number(1))



def find_prime_number(n):
  n = 1000
  array = [True for i in range(n + 1)]

  for i in range(2, int(math.sqrt(n)) + 1):
    if array[i] == True:
      j = 2
      while i * j <= n:
        array[i * j] = False
        j += 1

  for i in range(2, n + 1):
    if array[i]:
      print(i, end=' ')

#find_prime_number(100)

File: java/complete_search/prime_num.java
import java.util.*;

public class Solution {
    HashSet<Integer> numbersSet = new HashSet<>();

    public boolean isPrime(int num) {
        // 1. 0과 1은 소수가 아니다
        if (num == 0 || num == 1)
            return false;

        // 2. 에라토스테네스의 체의 limit 숫자를 계산한다.
        int lim = (int)Math.sqrt(num);

        // 3. 에라토스테네스의 체에 따라 lim까지 배수 여부를 확인한다.
        for (int i = 2; i <= lim; i++)
            if (num % i == 0)
                return false;

        return true;
    }

    public void recursive(String comb, String others) {
        //System.out.println("comb:" + comb);
        //System.out.println("others:" + others);
        // 1. 현재 조합을 set에 추가한다.
        if (!comb.equals(""))
            numbersSet.add(Integer.valueOf(comb));
        //System.out.println("numberSet:" + numbersSet);
        // 2. 남은 숫자 중 한 개를 더해 새로운 조합을 만든다.
        for (int i = 0; i < others.length(); i++) {
            //System.out.println("i=" + i);
            recursive(comb + others.charAt(i), others.substring(0, i) + others.substring(i + 1));
        }
        //System.out.println("");

    }

    public int solution(String numbers) {
        // 1. 모든 조합의 숫자를 만든다.
        recursive("", numbers);

        // 2. 소수의 개수만 센다.
        int count = 0;
        Iterator<Integer> it = numbersSet.iterator();
        while (it.hasNext()) {
            int number = it.next();
            if (isPrime(number))
                count++;
        }

        // 3. 소수의 개수를 반환한다.
        return count;
    }
}